### PR TITLE
Import syntax highlighting from Microsoft’s repo

### LINKED ISSUE #675: Reuse Grammars from Sublime-TypeScript
From https://github.com/TypeStrong/atom-typescript/pull/672, we should investigate if a simple npm script can be utilized to copy and repackage the grammars from Sublime.

## edits by @basarat: 

We have very little custom work ... 

* The top that needs to be preserved across versions: https://github.com/TypeStrong/atom-typescript/blob/729b5522cc17fc84511c05ebb5ae1812f3ec4b00/grammars/typescript.cson#L12-L80

* The angular template support
https://github.com/TypeStrong/atom-typescript/pull/798

### PR DESCRIPTION
**Pull request by blakeembrey**
Figured it's about time we close #675.

### CODE REVIEW

**Comment by blakeembrey** [comment]
More info: It's a little script that's pulling the raw files from GitHub, reading as YAML, adding the custom grammar rules before the other rules (in `expression`) and then writing as CSON to the grammar directory.

File: package.json [hunk 1]
```diff
@@ -66,6 +67,10 @@
     "xtend": "^4.0.0"
   },
   "devDependencies": {
+    "cson": "^3.0.2",
+    "js-yaml": "^3.5.2",
+    "mz": "^2.1.0",
+    "popsicle": "^3.0.1",
     "typings": "^0.2.3"
```
Comment (basarat) [inline package.json]: This is perfectly fine and thank you :+1: :100: . Just wanted to share that atom does install `devDependencies` on user machines for whatever reason. That's why we had to remove `grunt` / `grunt-ts`. Leaving these in shouldn't be a problem though :rose: Again just sharing past experiences :rose:

**Comment by basarat** [comment]
Amusing myself :rose: 

image

File: scripts/grammar.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+      },
+      '3': {
+        name: 'es6import.path.string'
+      }
+    }
+  }
+]
+
+Promise.all([
+  request('https://raw.githubusercontent.com/Microsoft/TypeScript-TmLanguage/master/TypeScript.YAML-tmLanguage'),
```
Comment (basarat) [inline scripts/grammar.js]: :rose: :100:

**Comment by basarat** [comment]
@blakeembrey merge or let me merge. Love you! :rose:

**Comment by blakeembrey** [comment]
Absolutely, all yours! On the Angular template, I wonder if there's a better way to detect HTML and switch to HTML within strings. Maybe if the string begins with `\\s*<`?

**Comment by basarat** [comment]
> Maybe if the string begins with \\s*<

would make a lot of people happy. I saw this yesterday: 

image

Very much not interested after TSX :rose:  Put me some html in ts and not ts in html.

**Comment by basarat** [comment]
Released https://github.com/TypeStrong/atom-typescript/releases/tag/v8.0.0 :rose:

**Comment by nycdotnet** [comment]
Way to go!

**Comment by tomitrescak** [comment]
Great!

**Pull request merged**